### DISCUSSION TITLE
prefix 'format_' for format related settings

### DISCUSSION THREAD

**Discussion by youngsofun** [databendlabs/databend#7935] [Ideas]
background： I`m planning to add more format setting/option

pros:

1.  compatible with clickhouse like https://clickhouse.com/docs/en/operations/settings/settings#json-formats-settings
2. they are only use din input/output format
3. make the meaning of setting name clear to user.
4. good to group them together when show settings.

cons:

1.  longer name.
2.  also apply to stream load, since it use settings.
5.  have to remember to  remove the 'format_'   prefix when used in stage option

**Comment by youngsofun** [comment]
cc @Xuanwo  @BohuTANG  @sundy-li

**Comment by bohutang** [comment]
I am not sure about the background more, is this useful for the case(?):
https://docs.snowflake.com/en/sql-reference/sql/create-file-format.html

**Comment by youngsofun** [comment]
I think we will support `create file format`
but It should be syntactic sugar on top of a set of format option/setting.

**Reply by bohutang** [reply]
Ok, I missunderstand.

**Comment by Xuanwo** [comment]
I support this change.

**Reply by sundy-li** [reply]
+1

**Comment by bohutang** [comment]
Done https://github.com/datafuselabs/databend/pull/7960
